File: proxads/src/main/java/com/proxglobal/proxads/adsv2/admax/openads/MaxOpenAdsApplication.java
package com.proxglobal.proxads.adsv2.admax.openads;

import android.app.Application;

public abstract class MaxOpenAdsApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        MaxOpenAds.getInstance().init(this, getOpenAdsId());
    }

    protected final void disableOpenAdsAt(Class... clss) {
        for (Class cls : clss) {
            MaxOpenAds.getInstance().registerDisableOpenAdsAt(cls);
        }
    }

    protected abstract String getOpenAdsId();
}

File: proxads/src/main/java/com/proxglobal/proxads/adsv2/admax/openads/MaxOpenAdsApplication.kt
package com.proxglobal.proxads.adsv2.admax.openads

import android.app.Application

abstract class MaxOpenAdsApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        MaxOpenAds.instance.init(this, getOpenAdsId())
    }

    protected fun disableOpenAdsAt(vararg clss: Class<*>?) {
        for (cls in clss) {
            MaxOpenAds.instance.registerDisableOpenAdsAt(cls!!)
        }
    }

    abstract fun getOpenAdsId(): String
}